### PR TITLE
Add CrashReporter with targetSettings to fixtures

### LINKED ISSUE #3488: [Dependencies.swift][SPM] Build Errors in PLCrashReporter Dependency
**Describe the bug**
*This was already briefly mentioned in #3286 but not investigated further yet.*

When importing [microsoft/plcrashreporter](https://github.com/microsoft/plcrashreporter) as a SPM dependency using Tuists Dependencies.swift, there are build errors in the dependencies source code.

**To Reproduce**

1. Use the attached minimal example zip
2. Run tuist dependencies fetch
3. Run tuist generate
4. Open the generated workspace
5. Build the main TuistPLCrashreporter target
6. Observe build errors in the CrashReporter target

Note that importing the same dependency using Xcodes SPM integration does not result in any build errors.

**Expected behavior**
I expect Tuist handled SPM dependencies to be compatible, when they are compatible with Xcode's built-in package management system.

**Desktop:**

- OS: macOS 11.5.2 / macOS 12 Developer Beta 6, Xcode 12.5.1
- Tuist Version: 1.49.2

[tuist-plcrashreporter.zip](https://github.com/tuist/tuist/files/7212417/tuist-plcrashreporter.zip)

### PR DESCRIPTION
**Pull request by ffittschen**
Resolves #3488
Request for comments document (if applies): -

### Short description 📝

As discussed in #3488, this PR adds the CrashReporter dependency with some `targetSettings` to the `app_with_spm_dependencies` fixtures.

### How to test the changes locally 🧐

1. Check out the project
2. Run `./fourier test tuist acceptance projects/tuist/features/dependencies.feature:11`

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 ### Added
 
 - Add ability to specify as a command line argument, the Xcode version to use when bundling/releasing tuist and its libraries [#3957](https://github.com/tuist/tuist/pull/3957) by [@hisaac](https://github.com/hisaac)
+- Add PLCrashReporter to fixture tests for iOS app with SPM dependencies [#4000](https://github.com/tuist/tuist/pull/3957) by [@ffittschen](https://github.com/ffittschen)
```
Comment (danyf90) [inline CHANGELOG.md:14]: I wouldn't add it to the CHANGELOG, as it's not a user facing change

File: projects/tuist/fixtures/app_with_spm_dependencies/Project.swift [hunk 2]
```diff
@@ -22,6 +22,7 @@ let project = Project(
                 .external(name: "FirebaseFirestore"),
                 .external(name: "GoogleSignIn"),
                 .external(name: "Realm"),
+                .external(name: "CrashReporter"),
```
Comment (danyf90) [inline projects/tuist/fixtures/app_with_spm_dependencies/Project.swift:25]: Could you please keep the alphabetical order? 🙏

**Review by danyf90 — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]
LGTM after addressing @danyf90's suggestions 👏

**Comment by ffittschen** [comment]
@danyf90 As #3996 has been merged, should I add these targetSettings to the `PackageInfoMapper.swift`, or should I just keep the `targetSettings` in the `app_with_spm_dependencies/Tuist/Dependencies.swift`?

**Comment by danyf90** [comment]
@ffittschen no strong opinion, you can keep it here for now

**Pull request merged**